packages/arb-validator-core/ethbridge/arbRollupWatcher.go

```diff
@@ -205,7 +258,37 @@ func (vm *ethRollupWatcher) processMessageDeliveredEvents(chainInfo arbbridge.Ch
 			Message:   msg,
 		}, nil
 
-	} else if ethLog.Topics[0] == ethDepositMessageDeliveredID {
+	case transactionBatchID:
+		tx, _, err := vm.client.TransactionByHash(ctx, ethLog.TxHash)
+		if err != nil {
+			return nil, err
+		}
+
+		var args TransactionBatchTxCallArgs
+		transactionBatchTxCallABI.Inputs.UnpackValues(tx.Data()[4:])
+		err = transactionBatchTxCallABI.Inputs.Unpack(&args, tx.Data()[4:])
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '4', create a named constant for it](https://app.codacy.com/gh/OffchainLabs/arbitrum/pullRequest?prid=5428340)